Dalamud.Injector/EntryPoint.cs

```diff
@@ -819,16 +829,57 @@ private static int ProcessLaunchCommand(List<string> args, DalamudStartInfo dala
                             RewriteRemoteEntryPointW(p.Handle, gamePath, JsonConvert.SerializeObject(startInfo)));
                         Log.Verbose("RewriteRemoteEntryPointW called!");
                     }
+
+                    if (!withoutDalamud && dalamudStartInfo.LoadMethod == LoadMethod.Hybrid)
+                    {
+                        DebugSetProcessKillOnExit(false);
+                        entryPoint = GetRemoteEntryPointW(p.Handle, gamePath);
+                        if (entryPoint == nint.Zero)
+                            throw new EntryPointNotFoundException("Couldn't get game process entry point");
+                        Log.Verbose("Got remote entry point {0}", entryPoint);
+                        var int3 = new byte[] { 0xCC };
+                        VirtualProtectEx(p.Handle, entryPoint, 1, MemoryProtection.ExecuteReadWrite, out var entryPointProtection);
+                        Log.Verbose("Original entry point protection {0}", entryPointProtection);
+                        ReadProcessMemory(p.Handle, entryPoint, entryPointInstruction, 1, out var numberOfBytesRead);
+                        Log.Verbose("Original entry point instruction {0}", entryPointInstruction[0]);
+                        WriteProcessMemory(p.Handle, entryPoint, int3, 1, out var numberOfBytesWritten);
+                        if (numberOfBytesWritten != 1)
+                            throw new Exception("Failed to set breakpoint at entry point");
+                        VirtualProtectEx(p.Handle, entryPoint, 1, entryPointProtection, out _);
+                        FlushInstructionCache(p.Handle, entryPoint, 1);
+                    }
+                },
+                (p, mt) =>
+                {
+                    if (!withoutDalamud && dalamudStartInfo.LoadMethod == LoadMethod.Hybrid)
+                    {
+                        var dwThreadId = WaitForBreakpointDebugEvent(entryPoint);
+                        VirtualProtectEx(p.Handle, entryPoint, 1, MemoryProtection.ExecuteReadWrite, out var entryPointProtection);
+                        WriteProcessMemory(p.Handle, entryPoint, entryPointInstruction, 1, out var numberOfBytesWritten);
+                        if (numberOfBytesWritten != 1)
+                            throw new Exception("Failed to restore entry point instruction");
+                        VirtualProtectEx(p.Handle, entryPoint, 1, entryPointProtection, out _);
+                        FlushInstructionCache(p.Handle, entryPoint, 1);
```

Comment:
https://www.codeproject.com/Articles/132742/Writing-Windows-Debugger-Part-2#RevertBP_at_SA

```cs
CONTEXT ctx;
GetThreadContext(mt, &ctx);
ctx.Rip--;
SetThreadContext(mt, &ctx);
```

is needed to decrement the instruction pointer back to where the INT3 was; otherwise, the instruction pointer will point to the second byte of the first instruction, resulting in `sub esp, 0x28` instead of `sub rsp, 0x28`, causing crash due to truncated stack pointer.

Reply:
Should be fixed now